lib/gitsh/git_repository.rb

```diff
@@ -8,7 +8,8 @@ def initialize(env)
     end
 
     def initialized?
-      git_dir && File.exist?(git_dir)
+      dir = git_dir
+      dir && File.exist?(dir)
```

Comment:
The `git_dir && …` in the original implementation appeared to be there to guard against `nil`, so I assumed that it could.

However, it seems I'm wrong. It can be the empty string, but since `File.exist?('')` returns `false` we can probably simplify this even further.